File: backend/src/main/java/com/sunrisedental/service/DentistService.java
package com.sunrisedental.service;

import com.sunrisedental.dto.DentistRequest;
import com.sunrisedental.dto.DentistResponse;
import com.sunrisedental.dto.PagedResponse;
import com.sunrisedental.entity.AvailabilityStatus;

import java.util.List;

/**
 * Service interface for Dentist management operations.
 */
public interface DentistService {

    DentistResponse create(DentistRequest request);

    PagedResponse<DentistResponse> getAll(
            String search,
            String specialization,
            AvailabilityStatus status,
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir);

    DentistResponse getById(Long id);

    DentistResponse update(Long id, DentistRequest request);

    void delete(Long id);

    DentistSummary getSummary();

    List<String> getSpecializations();

    List<DentistResponse> getActiveDentists();

    record DentistSummary(
            long total,
            long available,
            long busy,
            long onLeave
    ) {}
}

File: frontend/src/services/dentistService.js
import api from './api';

/**
 * dentistService — all API calls for Dentist Management.
 */
const dentistService = {

  getAll({ search = '', specialization = '', status = '', active = '', page = 0, size = 10, sortBy = 'createdAt', sortDir = 'desc' } = {}) {
    const params = { search, specialization, page, size, sortBy, sortDir };
    if (status) params.status = status;
    if (active !== '') params.active = active;

    return api.get('/dentists', { params });
  },

  getById(id) {
    return api.get(`/dentists/${id}`);
  },

  create(data) {
    return api.post('/dentists', data);
  },

  update(id, data) {
    return api.put(`/dentists/${id}`, data);
  },

  delete(id) {
    return api.delete(`/dentists/${id}`);
  },

  getSummary() {
    return api.get('/dentists/summary');
  },

  getSpecializations() {
    return api.get('/dentists/specializations');
  },

  getActiveDentists() {
    return api.get('/dentists/active');
  }
};

export default dentistService;
